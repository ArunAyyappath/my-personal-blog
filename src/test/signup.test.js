import React from "react";
import CustomSignUp from "../component/signUp";
import { configure, shallow, cleanup, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

afterEach = () => {
  cleanup;
};

describe("<CustomSignUp/>", () => {
  it("renders correctly", () => {
    const wrapper = renderer.create(<CustomSignUp />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("check-signup-function-call-times", () => {
    const wrapper = shallow(<CustomSignUp />);
    const spy = jest.spyOn(wrapper.instance(), "onclickHandle");
    const spyFnc = jest.spyOn(wrapper.instance(), "authListner");
    wrapper.setState({
      userIdSignUp: "arunayyappath@gmail.com",
      passwordIdSignUp: "123456"
    });
    wrapper.find("button").simulate("click");
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spyFnc).toHaveBeenCalledTimes(1);
  });

  it("verify-signin-userId", () => {
    const wrapper = mount(<CustomSignUp />);
    const container = wrapper.find("input[type='text']");
    container.find("input[type='text']").simulate("change", {
      target: { id: "userIdSignUp", value: "Fakeuser@gmail.com" }
    });
    expect(wrapper.instance().state.userIdSignUp).toBe("Fakeuser@gmail.com");
  });

  it("verify-signin-passwordId", () => {
    const wrapper = mount(<CustomSignUp />);
    const container = wrapper.find("input[type='text']");
    container.find("input[type='text']").simulate("change", {
      target: { id: "passwordIdSignUp", value: "12345678" }
    });
    expect(wrapper.instance().state.passwordIdSignUp).toBe("12345678");
  });
});
