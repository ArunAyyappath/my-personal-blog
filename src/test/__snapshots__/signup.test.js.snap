// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomSignUp/> renders correctly 1`] = `
Array [
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      className="makeStyles-background-1"
    >
      <div
        className="MuiAvatar-root makeStyles-ProfileP-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
          >
            <path
              d="M0 0h24v24H0V0z"
            />
            <path
              d="M0 0h24v24H0V0z"
              opacity=".87"
            />
          </g>
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Sign up
      </h1>
      <form
        className="makeStyles-formStyle-3"
        noValidate={true}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="userIdSignUp"
              >
                Email Address
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-213 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-214"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  autoComplete="email"
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="userIdSignUp"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="passwordIdSignUp"
              >
                Password
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-213 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-214"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  autoComplete="current-password"
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="passwordIdSignUp"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      style={
        Object {
          "paddingTop": "25px",
        }
      }
    >
      Confessions of a Writing Freak
    </p>
  </main>,
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <button
      className="submit m-4"
      onClick={[Function]}
      type="submit"
    >
      SignUp
    </button>
  </div>,
  <div
    className="home"
  >
    <a
      href="/"
    >
      Home
    </a>
  </div>,
  <div
    className="accOn"
  >
    <a
      href="/signIn"
    >
      Already have an account ? Sign In
    </a>
  </div>,
  <div />,
]
`;
